tasks/run_kops.yml

```diff
@@ -0,0 +1,178 @@
+---
+
+###
+### Diff State Store
+###
+- name: diff generated cluster config against remote state store
+  udiff:
+    source: "{{ lookup('template', 'cluster.yml.j2') }}"
+    target: |
+      {% if kops_aws_profile %}
+      # If using boto_profile, make kops aware of it
+      export AWS_PROFILE="{{ kops_aws_profile }}";
+      {% endif %}
+      # Don't fail to show that remote has nothing yet
+      kops get cluster -o yaml \
+        --state s3://{{ cluster.s3_bucket_name }} \
+        --name {{ cluster.name }} || true
+    source_type: string
+    target_type: command
+    diff: yaml
+    diff_yaml_ignore: ['creationTimestamp']
+  check_mode: False
+  register: _kops_diff_state_cluster
+  when:
+    - kops_update is defined
+    - kops_update == 'state' or kops_update == 'update' or kops_update == 'all'
+
+- name: diff generated instance group config against remote state store
+  udiff:
+    source: "{{ lookup('template', 'instance-groups.yml.j2') }}"
+    target: |
+      {% if kops_aws_profile %}
+      # If using boto_profile, make kops aware of it
+      export AWS_PROFILE="{{ kops_aws_profile }}";
+      {% endif %}
+      # Don't fail to show that remote has nothing yet
+      kops get instancegroups -o yaml \
+        --state s3://{{ cluster.s3_bucket_name }} \
+        --name {{ cluster.name }} || true
+    source_type: string
+    target_type: command
+    diff: yaml
+    diff_yaml_ignore: ['creationTimestamp']
+  check_mode: False
+  register: _kops_diff_state_instancegroups
+  when:
+    - kops_update is defined
+    - kops_update == 'state' or kops_update == 'update' or kops_update == 'all'
+
+- name: diff ssh public key against remote state store
+  udiff:
+    source: |
+      # Test SSH key for validity
+      if ! bash -c "ssh-keygen -lf <(echo '{{ lookup('template', 'ssh-key.pub.j2') }}')" >/dev/null; then
+        exit 1;
+      fi
+      # Get SSH key fingerprint only
+      bash -c "ssh-keygen -lE md5 -f <(echo '{{ lookup('template', 'ssh-key.pub.j2') }}')" \
+        | awk '{ print $2}' | sed 's/^MD5://g'
```

Comment:
depending on sed on the Ansible controller machine is not such a good idea. see differences between GNU sed vs BSD sed.

Reply:
@danvaida I found no compatibility issue as it is using a very basic search-and-replace with no fancy stuff.